build.ps1

```diff
@@ -84,124 +86,121 @@ Function CheckPublishTools()
-    cd (Join-Path $rootDir $path)
 
-    if ($clean) {
-        & dotnet clean --verbosity $verbosity --configuration $configuration
-        if ($LASTEXITCODE -ne 0) {
+    $projectPath = Join-Path $rootDir $path
+
+    if ($clean)
+    {
+        & dotnet clean $projectPath --verbosity $verbosity --configuration $configuration
+        if ($LASTEXITCODE -ne 0)
+        {
             throw "Clean failed: $label"
         }
     }
 
-    & dotnet build --verbosity $verbosity --configuration $configuration
+    & dotnet build $projectPath --verbosity $verbosity --configuration $configuration
 
-    if ($LASTEXITCODE -ne 0) {
+    if ($LASTEXITCODE -ne 0)
+    {
         throw "Build failed: $label"
     }
 }
 
-Function BuildPackage($path, $message) {
+Function BuildPackage($path, $message)
+{
     $label = "PACK: --- $message $configuration ---"
 
     Write-Host
     Write-Host -ForegroundColor Cyan $label
 
     $projectPath = Join-Path $rootDir $path
-    cd $projectPath
-
-    $projectName = (dir (Join-Path $projectPath *.csproj))[0].BaseName
+    $projectName = (Get-ChildItem (Join-Path $projectPath *.csproj))[0].BaseName
 
     if ($sign)
     {
-        Write-Host -ForegroundColor Magenta "`tSigning binaries: $projectName"
-        $filesToSign = dir -Recurse .\bin\Release\$projectName.dll
+        Set-Location $projectPath
+        Write-Host -ForegroundColor Magenta "`tSigning binaries for: $projectName"
+        $filesToSign = Get-ChildItem -Path "$projectPath\bin\$configuration\*\$projectName.dll" -Recurse
         SignDotNetBinary $filesToSign
     }
 
     & dotnet pack --verbosity $verbosity --configuration $configuration --no-build --include-symbols --include-source --output $localPackages
 
-    if ($LASTEXITCODE -ne 0) {
+    if ($LASTEXITCODE -ne 0)
+    {
         throw "Package failed: $label"
     }
 
     if ($sign)
     {
         Write-Host -ForegroundColor Magenta "`tSigning package: $projectName"
-        $filesToSign = dir (Join-Path $localPackages "$projectName.*.nupkg")
+        $filesToSign = Get-ChildItem (Join-Path $localPackages "$projectName.*.nupkg")
         SignNuGetPackage $filesToSign
     }
 }
 
-Function RunTests($path, $message, $framework="*", $filterTestCategory="*") {
-
-    if ($filterTestCategory -eq "*")
-    {
-        $label = "TEST: --- $message $configuration ---"
-    }
-    else
-    {
-        $label = "TEST: --- $message $configuration $filterTestCategory ---"
-    }
+Function RunTests($message, $framework = "*", $filterTestCategory = "*")
+{
+    $label = "TEST: --- $message $configuration $filterTestCategory ---"
 
     Write-Host
     Write-Host -ForegroundColor Cyan $label
-    cd (Join-Path $rootDir $path)
 
-    if ($framework -eq "*")
+    $testProjects = Get-ChildItem *tests.csproj -Recurse
+
+    $runTestCmd = "dotnet test --verbosity $verbosity --configuration $configuration --logger trx"
+    if ($filterTestCategory -ne "*")
     {
-        if ($filterTestCategory -eq "*")
```

Comment:
We can get rid of all these DRY violations by building a string of the command and running it